docs/features/nullable-reference-types.md

```diff
@@ -152,11 +152,11 @@ var x = (string)maybeNull; // x is string?, W warning
 var y = (string)oblivious; // y is string~, no warning
 var z = (string?)notNull; // y is string?, no warning
 ```
-Explicit casts change nested nullability.
-_Should a warning be reported when there is not an implicit nullable conversion (for instance `(List<object>)new List<object?>()`)?_
+A warning is reported if the type of operand including nested nullability is not explicitly convertible to the target type.
 ```c#
-List<object?> x = ...;
-var y = (List<object>)x; // y is List<object!>!, no warning
+var x =  new List<string>();
+var y = (List<string?>)x;        // warning
+var z = (IEnumerable<string?>)x; // no warning
```

Comment:
Is there a test for this scenario?

Reply:
Good catch. Added.

---
In reply to: [215080231](https://github.com/dotnet/roslyn/pull/29665#discussion_r215080231) [](ancestors = 215080231)